src/responder/common/cache_req/cache_req.c

```diff
@@ -1312,6 +1302,15 @@ static void cache_req_input_parsed(struct tevent_req *subreq)
         return;
     }
 
+    if (state->domain_name != NULL && domain != NULL
+        && strcmp(state->domain_name, domain) != 0) {
+        DEBUG(SSSDBG_CRIT_FAILURE,
+              "Mismatch between input domain name [%s] and parsed domain name [%s]\n",
+              state->domain_name, domain);
+        tevent_req_error(req, ERR_INPUT_PARSE);
+        return;
```

Comment:
This move is questionable. This check is also kind of "check for error".

(I didn't check if a mismatch really possible, but) in case this condition is met then `cache_req_set_name(state->cr, name)` is already done after this change.

Perhaps adding initializer ` char *domain = NULL;` would be better (if we rely on `sss_parse_inp_recv()` doesn't modify args in case it returns error).

Reply:
I see your point.

However, it is quite complex, as `sss_parse_inp_recv()` **does** modify domain when returning `ERR_DOMAIN_NOT_FOUND`. Let's go for the minimal change and adopt `char *domain = NULL`.